### DISCUSSION TITLE
Is there an OpenAPI Spec or other definition of the REST API?

### DISCUSSION THREAD

**Discussion by Splaktar** [verdaccio/verdaccio#2539] [Q&A] [answered]
I dug through the code a bit and I saw lots of Express set up for various APIs, but I couldn't find the exact API that maps to `npm info`. I know that there's a sidebar endpoint that I could use, but it returns a lot more than what I need.

It would be great if there was a REST API definition that I could search through. I checked the docs, issues, etc and I couldn't find any mention of one. Maybe I missed it?

**Comment by juanpicado** [comment, marked as answer]
hi @Splaktar  no docs from our side, but the closest thing is https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md 
at verdaccio yeah we follow that guidance (because it is what all (registries vendors + package managers) are doing) and furthermore we have some web endpoints custom for verdaccio which are not meant to be used outside of the web.

In the last months and most likely the next ones I am working on clean up a lot of things, refactor and hopefully, this bring more clear guidance on how to use or understand internal endpoints. 

Something like this I guess would be great to have.
eg: https://github.com/apidoc/apidoc 
https://apidocjs.com/example/#api-User-DeleteUser

**Reply by Splaktar** [reply]
Thanks! https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md is very helpful.

**Reply by juanpicado** [reply]
There is an oldie API also here https://github.com/npm/npm-registry-couchapp/blob/master/registry/rewrites.js
